src/client/actions/GuildBanAdd.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const Action = require('./Action');
+const { Events } = require('../../util/Constants');
+
+class GuildBanAdd extends Action {
+  handle(data) {
+    const client = this.client;
+    const guild = client.guilds.cache.get(data.guild_id);
+
+    /**
+     * Emitted whenever a member is banned from a guild.
+     * @event Client#guildBanAdd
+     * @param {GuildBan} ban The ban that occurred
+     */
+    if (guild) {
+      client.emit(Events.GUILD_BAN_ADD, guild.bans.add({ user: data.user, reason: data.reason }));
```

Comment:
```suggestion
      client.emit(Events.GUILD_BAN_ADD, guild.bans.add({ user: data.user }));
```
you can only get the reason by fetching the ban 👀 ([reference](https://discord.com/developers/docs/topics/gateway#guild-ban-add-guild-ban-add-event-fields))

Reply:
this is intentional to make the ban object partial

Reply:
can be done now after the latest commit

Reply:
done in 1e834c2 since the button to do it automagically wasn't there